### DISCUSSION TITLE
Filterbar mit mehreren Icons

### DISCUSSION THREAD

**Discussion by muhammed-basaran** [neuefische/questions#441] [Web] [answered]
Hallo zusammen, 
ich habe eine Filterbar erstellt, die aus mehreren Icons besteht. Zunächst möchte ich eine einfache Eigenschaft einbauen, wenn man auf eines dieser Icons klickt, soll dieser hervorgehoben werden ( sprich die Hintergrundfarbe oder die Schriftfarbe soll sich ändern). Aktuell hab ich das Problem, wenn ich auf ein Icon klicke, ändern sich alle Icons. Wie muss ich denn vorgehen, um den Effekt nur auf das betroffene Icon zu begrenzen? 

Aktuell sieht es folgendermaßen aus: 

<img width="603" alt="Bildschirmfoto 2023-01-06 um 20 32 34" src="https://user-images.githubusercontent.com/115261964/211085960-c6f06689-f4ed-4acb-a67f-45b10bd4d3c4.png">

<img width="523" alt="Bildschirmfoto 2023-01-06 um 20 36 15" src="https://user-images.githubusercontent.com/115261964/211086497-25952dab-8b29-4e9a-8c45-8e59707b2175.png">

<img width="508" alt="Bildschirmfoto 2023-01-06 um 20 33 57" src="https://user-images.githubusercontent.com/115261964/211086522-c5939cb5-e156-41cb-afc2-758d77ef5715.png">

Danke schon mal für eure Hilfe und Vorschläge!

**Comment by actyralf** [comment, marked as answer]
Just having a variable `isClicked` is not enough. You need to have a state variable telling you which icon has been selected, e.g.:

```js
const [selectedIcon, setSelectedIcon]=useState(null);

// ...

<StyledBrandNike
   onClick = {()=>{setSelectedIcon("nike")}}
   color = {selectedIcon==="nike"?"gray":""}
/>

// and so on ...

```

Alternatively you could use the id or any other unique identifier.

**Reply by muhammed-basaran** [reply]
thanks for your help !!! finally it works 🙏
